lightly/data/lighty_subset.py

```diff
@@ -0,0 +1,37 @@
+from typing import List, Dict, Tuple
+
+from lightly.data.dataset import LightlyDataset
+
+
+class LightlySubset(LightlyDataset):
+    def __init__(self, base_dataset: LightlyDataset, filenames_subset: List[str]):
+        self.base_dataset = base_dataset
+
+        dict_base_dataset_filename_index: Dict[str, int] = \
+            dict([
+                (base_dataset.index_to_filename(self.dataset, index), index)
+                for index in range(len(base_dataset))
+            ])
+        self.mapping_subset_index_baseset = \
```

Comment:
I'd rename it to be more clear: `subset_index_to_baseset_index`, `subset_index_to_dataset_index`, `map_subset_index_to_baseset_index` are possible candidates.